lib/resources/oracledb_session.rb

```diff
@@ -13,30 +16,110 @@ class OracledbSession < Inspec.resource(1)
       end
     "
 
-    attr_reader :user, :pass, :host, :sid, :sqlplus_bin
+    attr_reader :user, :password, :host, :service
 
     def initialize(opts = {})
       @user = opts[:user]
-      @pass = opts[:pass]
+      @password = opts[:password] || opts[:pass]
+      if opts[:pass]
+        warn '[DEPRECATED] use `password` option to supply password instead of `pass`'
+      end
+
       @host = opts[:host] || 'localhost'
-      @sid = opts[:sid]
+      @port = opts[:port] || '1521'
+      @service = opts[:service]
+
+      # we prefer sqlci although it is way slower than sqlplus, but it understands csv properly
+      @sqlcl_bin = 'sql'
       @sqlplus_bin = opts[:sqlplus_bin] || 'sqlplus'
-      return skip_resource("Can't run Oracle checks without authentication") if @user.nil? or @pass.nil?
+
+      return skip_resource "Can't run Oracle checks without authentication" if @user.nil? or @password.nil?
+      return skip_resource 'You must provide at least an SID and/or a Service Name for the session' if @service.nil?
     end
 
     def query(q)
       escaped_query = q.gsub(/\\/, '\\\\').gsub(/"/, '\\"')
-      cmd = inspec.command("echo \"#{escaped_query}\" | #{@sqlplus_bin} -s #{@user}/#{@pass}@#{@host}/#{@sid}")
+      # escape tables with $
+      escaped_query = escaped_query.gsub('$', "\\$")
+
+      p = nil
+      # check if sqlcl is available and prefer that
+      if inspec.command(@sqlcl_bin).exist?
+        bin = @sqlcl_bin
+        opts = "set sqlformat csv\nSET FEEDBACK OFF"
+        p = 'parse_csv_result'
```

Comment:
Let's use a better variable name here, such as `parse_method`, and since we're already going to convert it to a symbols, let's set it as a symbol here and avoid the indirection.

```
parse_method = :parse_csv_result
```